CheckpointPlugin.cpp

```diff
@@ -629,25 +711,53 @@ bool CheckpointPlugin::rewind(ServerWrapper sw) {
 	}
 	rewindState.buttonsDown = buttonsDown;
 
-	// Determine how much to rewind / advance time.
-	if (abs(ci.Steer) < .05f) { // Ignore slight input; keep current game state.
+	
+	// Ignore slight input; keep current game state.
+	const float rewindThreshold = cvarManager->getCvar("rewind_threshold").getFloatValue();
+	if (abs(rewindInput) < rewindThreshold) {
 		return true; // Ignoring input; apply state.
 	}
+
 	rewindState.deleting = false;
-	if (ci.Steer < -.95 && rewindState.holdingFor <= 0) {
+
+	// Determine how much to rewind / advance time.
+	if (rewindInput < -.95 && rewindState.holdingFor <= 0) {
 		rewindState.holdingFor -= elapsed;
-	} else if (ci.Steer > .95 && rewindState.holdingFor >= 0) {
+	} else if (rewindInput > .95 && rewindState.holdingFor >= 0) {
 		rewindState.holdingFor += elapsed;
 	} else {
 		rewindState.holdingFor = 0;
 	}
+
+	// Normalize rewindInput based on the threshold
+	float excessInput = std::abs(rewindInput) - rewindThreshold;
+	float normalRewindInput;
+	if (excessInput > 0) {
+		// Normalize only if rewindInput exceeds the threshold
+		float normalizedInput = excessInput / (1.0 - rewindThreshold);
+		normalRewindInput = (rewindInput < 0 ? -1 : 1) * normalizedInput;
+	} else {
+		normalRewindInput = 0;  // No effective input if below threshold
+	}
+
 	float factor = std::clamp(abs(rewindState.holdingFor) * 2, 1.0f, 10.0f);
 
 	// How much (in seconds) to move "current" (positive or negative)
-	float deltaElapsed = factor * elapsed * ci.Steer; // full left = 2-5 seconds/second
+	float deltaElapsed = factor * elapsed * normalRewindInput; // full left = 2-5 seconds/second
+
+	// if you are trying to use the matching axis more than the rewind axis
+	// and you've haven't hit the threshold to unpause, don't rewind
+	if ((matchingAxis != "None" && 2 * fabs(matchingInput) > fabs(rewindInput))) {
+		deltaElapsed = 0;
+	}
+
+	if (deltaElapsed != 0) {
+		rewindState.atCheckpoint = false;
+	}
```

Comment:
this fixes the bug where it still thinks you're at a checkpoint while rewinding, causing you to reset to that checkpoint instead of the place you rewinded to

Reply:
introduced by my changes, where you could rewind after a checkpoint load. i will put it back to the original behavior, where it unpauses if you press anything while at a checkpoint

Reply:
this also has interactions with the ability to do multiple checkpoints above. i will split all that out as a new feature branch